app/tasks/maintenance_tasks/array_task.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module MaintenanceTasks
+  class ActiveRecordTask < Task
+    # TODO: define replacement for Task.collection, delegating to instance
+    # Probably simplest to make this an ActiveSupport::Concern
+
+    # TODO: Specify abstract_class
+
+    EnumeratorBuilder = Struct.new(:collection) do
+      def enumerator(context:)
+        enumerator_builder.array(collection, cursor: context.cursor)
+      end
+    end
+
+    def enumerator(context:)
+      collection = self.collection
+      assert_array!(collection)
+
+      EnumeratorBuilder.new(collection)
+    end
+
+    def collection
```

Comment:
With separate sub-classes, maybe it makes sense to rename the methods:

- `ActiveRecordTask#collection` -> `ActiveRecordTask#relation`
- `ArrayTask#collection` -> `ArrayTask#array`